src/Compilers/Core/Portable/InternalUtilities/ReadOnlyMemoryOfCharComparer.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal sealed class ReadOnlyMemoryOfCharComparer : IEqualityComparer<ReadOnlyMemory<char>>
+    {
+        public static ReadOnlyMemoryOfCharComparer Instance { get; } = new();
```

Comment:
💡 Compilers should probably add/enable an analyzer for this, since it will not be obvious to people that even the clearest wins like this one are disallowed.

Reply:
Agree on analyzer.  Also fixed this up.